modules/client/src/internal/client/methods.ts

```diff
@@ -277,76 +279,116 @@ export class ClientMethods extends ClientCore implements IClientMethods {
   ): AsyncGenerator<DaoDepositStepValue> {
```

Comment:
Isn't this function becoming way too large? I would recommend splitting it into smaller chunks

Reply:
yep maybe it is too big right now